Cython/Utility/ModuleSetupCode.c

```diff
@@ -622,6 +622,14 @@ static CYTHON_INLINE void * PyThread_tss_get(Py_tss_t *key) {
   #define __Pyx_PyNumber_InPlaceDivide(x,y)  PyNumber_InPlaceDivide(x,y)
 #endif
 
+#if PY_MAJOR_VERSION >= 3 && (!CYTHON_COMPILING_IN_PYPY || PYPY_VERSION_NUM >= 0x07010100)
+#define __Pyx_PyDict_GetItemWithError PyDict_GetItemWithError
+#define __Pyx_PyDict_GetItemWithError_ErrOccurred PyErr_Occurred
+#else
+#define __Pyx_PyDict_GetItemWithError PyDict_GetItem
```

Comment:
That's the thing – this function isn't equivalent to `PyDict_GetItemWithError()`, because it does not properly report all errors (i.e. exceptions). Thus the actual call to `d.get()` in `Optimize.c`, for example. The `WithError` C-API function was added later for exactly this reason.

Also, such a change definitely shouldn't be part of an "enable pypy in tests" PR. :)

Reply:
Yes that makes sense. I'll get rid of this.